City-of-Helsinki/drupal-hdbt
# UHF-10752 Farewell chunks

# [UHF-10752](https://helsinkisolutionoffice.atlassian.net/browse/UHF-10752)


## What was done


* Use `splitChunks: false` for `npm run build`
   * This won't actually do anything unless we explicitly try to use chunking
* Converted async imports to static imports in `nav-toggle.js`.
   * If there are dynamic imports in any of the `./src/js/` files, Webpack will always create chunks, even if the splitChunks is set false. Dynamic imports are designed to create separate chunks by definition. 
   * This should fix the chunk load errors, like: `Loading chunk 305 failed. (error: hdbt/dist/js/async/toggleWidgets.chunk.js)`

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-10752`
* Run `make drush-cr`

## How to test


* [x] Check that the following functionalities still work in mobile and desktop. 
   * [x] Click on the search button, "other languages" button and menu button. Also test that the menu works as before.  (notice that when the code works as it should, url does not change when clicking these buttons, url change is the fallback option)
   Keep the console open, so you'll see possible errors. 
![image](https://github.com/user-attachments/assets/a2e638df-f0d2-4c46-a4d2-848075c82d25)



[UHF-10752]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-10752?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
